pkg/cmd/cli/uninstall/uninstall.go

```diff
@@ -293,10 +343,53 @@ func gracefullyDeleteRestore(ctx context.Context, kbClient kbclient.Client, name
 	return err
 }
 
-func forcedlyDeleteRestore(ctx context.Context, kbClient kbclient.Client, namespace string) error {
+func gracefullyWaitDataUpload(ctx context.Context, kbClient kbclient.Client, namespace string) error {
```

Comment:
Creating each function for Restore, DataUpload, and DataDownload seems a bit redundant.
Could we use a unified function for all three resources?

Reply:
I've used the `reflect` mechanism in Golang to restructure it.